presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraSplitManager.java

```diff
@@ -176,7 +180,7 @@ private static String buildTokenCondition(String tokenExpression, String startTo
                     size++;
                     if (size > partitionSizeForBatchSelect) {
                         String partitionId = String.format("%s in (%s)", partitionKeyColumnName, sb.toString());
-                        CassandraSplit split = new CassandraSplit(connectorId, schema, table, partitionId, null, hostMap.get(entry.getKey()));
+                        CassandraSplit split = new CassandraSplit(connectorId, schema, table, partitionId, clusteringKeysPredicate, hostMap.get(entry.getKey()));
```

Comment:
Make sure that cassandra supports queries like `partition_key IN <value> AND clustering_key = <value>`